lightning/src/chain/keysinterface.rs

```diff
@@ -346,6 +346,13 @@ pub trait KeysInterface: Send + Sync {
 	/// onion packets and for temporary channel IDs. There is no requirement that these be
 	/// persisted anywhere, though they must be unique across restarts.
 	fn get_secure_random_bytes(&self) -> [u8; 32];
+
+	/// Reads a `ChanKeySigner` for this `KeysInterface` from the given input stream.
+	/// This is only called during deserialization of other objects which contain
+	/// `ChannelKeys`-implementing objects (ie `ChannelMonitor`s and `ChannelManager`s) and must
+	/// read exactly the bytes that `<Self::ChanKeySigner as Writeable>::write()` writes, or return
```

Comment:
What if you upgrade/downgrade `KeysInterface` between r/w ? An older implementation may discard some data chunks written by a newer implementation if version is done correctly but I think your requirement scope this ?

Reply:
Maybe say instead "should validate that the length of the data is as expected".

Reply:
I added a note about versioning and extra bytes.